lib/immobilienscout/request_error_handler.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Immobilienscout
+  module RequestErrorHandler
+    class << self
+      ERROR_RESOURCE_NOT_FOUND = 'ERROR_RESOURCE_NOT_FOUND'
+      ERROR_COMMON_RESOURCE_NOT_FOUND = 'ERROR_COMMON_RESOURCE_NOT_FOUND'
+      ERROR_RESOURCE_VALIDATION = 'ERROR_RESOURCE_VALIDATION'
+
+      def handle(parsed_response)
+        if parsed_response.messages.is_a?(Array)
+          error_code = parsed_response.messages&.first&.code
+          error_messages = parsed_response.messages.map(&:messages)
+        else
+          error_code = parsed_response.messages['messageCode']
+          error_messages = Array.wrap(parsed_response.messages['message'])
+        end
+
+        error_class = error_class(error_code)
+        raise error_class, error_messages
+      end
+
+      private
+
+      def error_class(error_code)
+        case error_code
+        when ERROR_RESOURCE_NOT_FOUND
+          Immobilienscout::Errors::ResourceNotFound
+        when ERROR_COMMON_RESOURCE_NOT_FOUND
+          Immobilienscout::Errors::CommonResourceNotFound
+        when ERROR_RESOURCE_VALIDATION
+          Immobilienscout::Errors::ResourceValidation
+        else
+          Immobilienscout::Errors::InvalidRequest
```

Comment:
We are working on Homeday on the Stats Job file rescuing from `Immobilienscout::Errors::InvalidRequest` when a property is not found, and since is not returning this anymore but `Immobilienscout::Errors::ResourceNotFound` the change should also go there

Reply:
Hey @angela4713 , a rescue of `Immobilienscout::Errors::InvalidRequest` still works since the new errors inherit from this error. No breaking changes 😉